### PR TITLE
Remove IP address information from telemetry

### LINKED ISSUE #3139: Disable the collection of geolocation data in ApiView
Currently we do not collect IP address info, however AppInsights uses the IP address (at request time) to look up the location the request is coming from.  AppInsights does this by default. We do not want to collect that information for ApiView either.

Privacy review tracking [issue](https://global.azure.com/privacymanager/service/4405e061-966a-4249-afdd-f7435f54a510/review/DataTaggingTab).
Instructions on how to disable this: [link](https://stackoverflow.com/questions/64382237/how-can-i-avoid-collecting-client-location-info-from-application-insights-in-nod)

### PR DESCRIPTION
**Pull request by praveenkuttappan**
Remove IP address from app insights telemetry to disable collecting Geo info.

Fixes #3139

### CODE REVIEW

File: src/dotnet/APIView/APIViewWeb/Filters/TelemetryIpAddressFilter.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        private ITelemetryProcessor Next { get; set; }
+
+        public TelemetryIpAddressFilter(ITelemetryProcessor next)
+        {
+            this.Next = next;
+        }
+
+        public void Process(ITelemetry item)
+        {
+            if(item.Context?.Location?.Ip != null)
```
Comment (JoshLove-msft) [inline src/dotnet/APIView/APIViewWeb/Filters/TelemetryIpAddressFilter.cs:18]:
```suggestion
            if (item.Context?.Location?.Ip != null)

```

File: src/dotnet/APIView/APIViewWeb/Filters/TelemetryIpAddressFilter.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.ApplicationInsights.DataContracts;
+
+namespace APIViewWeb.Filters
+{
+    public class TelemetryIpAddressFilter : ITelemetryProcessor
+    {
+        private ITelemetryProcessor Next { get; set; }
```
Comment (JoshLove-msft) [inline src/dotnet/APIView/APIViewWeb/Filters/TelemetryIpAddressFilter.cs:9]:
```suggestion
        private readonly ITelemetryProcessor _next;

```

File: src/dotnet/APIView/APIViewWeb/Filters/TelemetryIpAddressFilter.cs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        }
+
+        public void Process(ITelemetry item)
+        {
+            if(item.Context?.Location?.Ip != null)
+            {
+                item.Context.Location.Ip = null;
+            }
+
+            this.Next.Process(item);
```
Comment (JoshLove-msft) [inline src/dotnet/APIView/APIViewWeb/Filters/TelemetryIpAddressFilter.cs:23]:
```suggestion
            _next.Process(item);

```

**Review by JoshLove-msft — APPROVED** [review]

**Review by benbp — APPROVED** [review]
LGTM after Josh's suggestions.

**Pull request merged**